AK/String.h

```diff
@@ -92,10 +92,18 @@ class String {
     // Creates a new String with a single code point repeated N times.
     static ErrorOr<String> repeated(u32 code_point, size_t count);
 
+    // NOTE: This duplicates what is in LibUnicode - but we do not want to drag that into this header.
```

Comment:
This duplication leaves us in a pretty awkward state at the end of this PR because the definition in LibUnicode is essentially unused after the case transformations are removed from CharacterTypes.h.

How about instead we define this enum in AK/StringUtils.h (since StringView may also want this definition in the future) and export it with a `using` declaration at the bottom of the header. And make LibUnicode use this enum definition instead of its own.